vpr/test/test_vpr.cpp

```diff
@@ -188,21 +188,21 @@ TEST_CASE("read_rr_graph_metadata", "[vpr]") {
     std::mt19937 g(1);
     std::shuffle(src_order.begin(), src_order.end(), g);
 
-    CHECK(device_ctx.rr_node_metadata.size() == 1);
-    CHECK(device_ctx.rr_edge_metadata.size() == 1);
+    CHECK(device_ctx.rr_graph_builder.rr_node_metadata_size() == 1);
+    CHECK(device_ctx.rr_graph_builder.rr_edge_metadata_size() == 1);
 
     auto node = arch.strings.intern_string(vtr::string_view("node"));
     auto edge = arch.strings.intern_string(vtr::string_view("edge"));
 
-    for (const auto& node_meta : device_ctx.rr_node_metadata) {
+    for (const auto& node_meta : device_ctx.rr_graph_builder.rr_node_metadata_) {
         CHECK(src_order[node_meta.first] == src_inode);
         REQUIRE(node_meta.second.has(node));
         auto* value = node_meta.second.one(node);
         REQUIRE(value != nullptr);
         CHECK_THAT(value->as_string().get(&arch.strings), Equals("test node"));
     }
 
-    for (const auto& edge_meta : device_ctx.rr_edge_metadata) {
+    for (const auto& edge_meta : device_ctx.rr_graph_builder.rr_edge_metadata_) {
```

Comment:
This is the GCC error.
/home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/test/test_vpr.cpp:205:79: error: passing ‘const RRGraphBuilder’ as ‘this’ argument discards qualifiers [-fpermissive]
for (const auto& edge_meta : device_ctx.rr_graph_builder.rr_edge_metadata()) {
^
In file included from /home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/src/device/rr_graph_view.h:4,
from /home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/src/base/vpr_context.h:12,
from /home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/src/base/globals.h:8,
from /home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/src/base/vpr_api.h:34,
from /home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/test/test_vpr.cpp:8:
/home/users/raza.jafari/Desktop/API_TASK/vtr-verilog-to-routing/vpr/src/device/rr_graph_builder.h:46:51: note: in call to ‘MetadataStorage<std::tuple<int, int, short int> >& RRGraphBuilder::rr_edge_metadata()’
MetadataStorage<std::tuple<int, int, short>>& rr_edge_metadata();
This error comes when I call the method for returning rr_node_metadata_